src/storage/ExtensionBasedMapper.ts

```diff
@@ -0,0 +1,129 @@
+import { posix } from 'path';
+import { types } from 'mime-types';
+import { RuntimeConfig } from '../init/RuntimeConfig';
+import { ResourceIdentifier } from '../ldp/representation/ResourceIdentifier';
+import { APPLICATION_OCTET_STREAM, TEXT_TURTLE } from '../util/ContentTypes';
+import { ConflictHttpError } from '../util/errors/ConflictHttpError';
+import { NotFoundHttpError } from '../util/errors/NotFoundHttpError';
+import { trimTrailingSlashes } from '../util/Util';
+import { FileIdentifierMapper } from './FileIdentifierMapper';
+
+const { join: joinPath, normalize: normalizePath } = posix;
+
+export interface PathAbstraction {
+
+  /**
+   * The parent directory path of the file.
+   */
+  path: string;
+
+  /**
+   * The name/slug of the file.
+   */
+  slug?: string;
+}
+
+export class ExtensionBasedMapper implements FileIdentifierMapper {
+  private readonly runtimeConfig: RuntimeConfig;
+  private readonly types: Record<string, any>;
+
+  public constructor(runtimeConfig: RuntimeConfig, overrideTypes = { acl: TEXT_TURTLE, metadata: TEXT_TURTLE }) {
+    this.runtimeConfig = runtimeConfig;
+    this.types = { ...types, ...overrideTypes };
+  }
+
+  // Using getters because the values of runtimeConfig get filled in at runtime (so they are still empty at
+  // construction time until issue #106 gets resolved.)
+  public get baseRequestURI(): string {
+    return trimTrailingSlashes(this.runtimeConfig.base);
+  }
+
+  public get rootFilepath(): string {
+    return trimTrailingSlashes(this.runtimeConfig.rootFilepath);
+  }
+
+  /**
+   * Strips the baseRequestURI from the identifier and checks if the stripped base URI matches the store's one.
+   * @param identifier - Incoming identifier.
+   *
+   * @throws {@link NotFoundHttpError}
+   * If the identifier does not match the baseRequestURI path of the store.
+   *
+   * @returns Absolute path of the file.
+   */
+  public mapUrlToFilePath(identifier: ResourceIdentifier, id = ''): string {
+    return this.getAbsolutePath(this.getRelativePath(identifier), id);
+  }
+
+  /**
+   * Strips the rootFilepath path from the filepath and adds the baseRequestURI in front of it.
+   * @param path - The file path.
+   *
+   * @throws {@Link Error}
+   * If the file path does not match the rootFilepath path of the store.
+   *
+   * @returns Url of the file.
+   */
+  public mapFilePathToUrl(path: string): string {
+    if (!path.startsWith(this.rootFilepath)) {
+      throw new Error(`File ${path} is not part of the file storage at ${this.rootFilepath}.`);
+    }
+    return this.baseRequestURI + path.slice(this.rootFilepath.length);
+  }
+
+  /**
+   * Get the content type from a file path, using its extension.
+   * @param path - The file path.
+   *
+   * @returns Content type of the file.
+   */
+  public getContentTypeFromExtension(path: string): string {
```

Comment:
Let's an issue that tracks the removal of this one from the _public_ interface.

Reply:
So `getContentTypeFromExtension` should likely not be on the public interface, because the only one who is allowed to know that `ExtensionBasedMapper` determines the content type from the extension is `ExtensionBasedMapper` itself.